addons/binding/org.openhab.binding.konnected/src/main/java/org/openhab/binding/konnected/handler/KonnectedHandler.java

```diff
@@ -0,0 +1,330 @@
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.StateDescription;
+import org.openhab.binding.konnected.internal.KonnectedConfiguration;
+import org.openhab.binding.konnected.internal.KonnectedDynamicStateDescriptionProvider;
+import org.openhab.binding.konnected.internal.KonnectedHTTPUtils;
+import org.openhab.binding.konnected.internal.KonnectedPutSettingsTimer;
+import org.openhab.binding.konnected.internal.servelet.KonnectedHTTPServelet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link KonnectedHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Zachary Christiansen - Initial contribution
+ */
+
+public class KonnectedHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(KonnectedHandler.class);
+
+    private KonnectedConfiguration config;
+    private KonnectedHTTPServelet webHookServlet;
+    private KonnectedPutSettingsTimer putSettingsTimer;
+    private List<String> sensors;
+    private List<String> actuators;
+    private List<Boolean> isAct;
+    KonnectedHTTPUtils http;
+    private KonnectedDynamicStateDescriptionProvider dynamicStateDescriptionProvider;
+
+    public KonnectedHandler(Thing thing, KonnectedHTTPServelet webHookServlet,
+            KonnectedDynamicStateDescriptionProvider provider) {
+        super(thing);
+        this.webHookServlet = webHookServlet;
+        this.putSettingsTimer = new KonnectedPutSettingsTimer();
+        this.sensors = new LinkedList<String>();
+        this.actuators = new LinkedList<String>();
+        http = new KonnectedHTTPUtils();
+        this.isAct = new LinkedList<Boolean>();
+
+        dynamicStateDescriptionProvider = provider;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // get the zone number in integer form
+        Integer zone = Integer.parseInt(channelUID.getId().substring((channelUID.getId().length() - 1)));
+
+        // convert the zone to the pin based on value at index of zone
+        Integer pin = Arrays.asList(PIN_TO_ZONE).get(zone);
+        String scommand = command.toString();
+
+        if (scommand.endsWith("N") || scommand.endsWith("1")) {
+            scommand = "1";
+        } else if (scommand.endsWith("F") || scommand.endsWith("0")) {
+            scommand = "0";
+        } else {
+            logger.debug("The command string was not proper setting to off fix your item");
```

Comment:
So since i am using a string item type, this is to log if a command attempts to send a string that is not a 1 or 0.  Perhaps i should reword the error message?